drafts/cypress-shallow-component-tests.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Shallow Component Tests in Cypress with Angular Standalone Components
+subtitle: Dynamically Overriding and Mocking Standalone Component Imports
+domain: software-engineering-corner.hashnode.dev
+tags: angular, components, testing
+cover: https://cdn.hashnode.com/res/hashnode/image/stock/unsplash/nXKNn2L4fDw/upload/150adf478d431d819cf7f88e3eaa92f7.jpeg?w=1600&h=840&fit=crop&crop=entropy&auto=compress,format&format=webp
+publishAs: timouti
+hideFromHashnodeCommunity: false
+ignorePost: true
+---
+We're all familiar with the testing pyramid principle, advocating for an abundance of unit tests and 
+only a handful of e2e tests.
+In frontend applications, the emphasis lies on numerous tests validating our component classes and 
+services, with only a sparse selection of e2e tests directly engaging with the UI. 
+However, a component is more than just its class. While class-only tests confirm behavior, 
+they may fall short in assessing interactions with the DOM, and even fail to guarantee accurate rendering.
+
+Angular 17 releases new built-in control flows transferring more logic to the component templates.
+Consequently, the urge to validate component templates and interactions is on the rise, 
+potentially surpassing the significance of unit tests.
+
+Fortunately, Cypress offers a solution in the form of "component tests," applicable to Angular, React, Vue, and Svelte. 
+For the sake of simplicity, we will delve into Angular component tests in this article.
+
+### Cypress Component Tests
+The objective of component tests is to validate and interact with the rendered component in the browser. 
+Unlike conventional e2e tests, these tests operate in isolation and are thus "shallow", just like standard unit test.
+
+To achieve this, certain providers and imports of the component need to be mocked. 
+Cypress facilitates this through `cy.mount` with an optional `MountConfig` parameter. 
+This `MountConfig` empowers the definition and override of component imports, providers, as well as 
+input and output properties. 
+
+```typescript
+beforeEach(() => {
+  cy.mount(MyComponent, {
+      imports: [
+          MyChildComponent, 
+          RouterTestingModule,
+          HttpClientTestingModule,
+      ], 
+      providers: [{ provide: MyService, useClass: MyMockService }]
+  });
+});
+```
+By leveraging the `MountConfig`, mocking providers and components becomes straightforward using your own 
+MockClass. However, in the context of Angular standalone components, where imports and providers are 
+integral part of the component itself, overriding child components via `MountConfig` encounters limitations.
+
+### Overriding Standalone Child Component
+The easiest solution involves mocking components with third-party libraries like `ng-mocks`. 
```

Comment:
Did you actively decide against giving an example here? Would be the fast-track for a lot of other projects I guess